### DISCUSSION TITLE
How to Change the MAC Address of the Lenovo ThinkSmart View - CD-18781Y

### DISCUSSION THREAD

**Discussion by mllockwood** [dinki/View-Assist#6] [Q&A] [answered]
Having the same MAC address for all 8 of my ThinkSmart units wreaks havoc for DHCP and my subnet.  Even if an IP address is hard coded in to the unit, Fully Kiosk will not work since it references the MAC address as well.

**Comment by dinki** [comment]
I've had success using this to change the MAC address on my second TS device.  It's a bit more command line driven so if you need help reach out.

https://community.home-assistant.io/t/is-this-the-perfect-standalone-tablet-for-ha/658422/581

**Comment by mllockwood** [comment, marked as answer]
Change MAC address of WiFi interface on the Lenovo ThinkSmart device:

(These instructions assume you already have the Windows applications installed and setup for ADB commands)

- Make sure the USB cable is connected between the Lenovo device and your PC.
- Open the ADB and Fastboot ++ program in Windows (already open from above steps)
- Create a folder in Windows. I created it in C:\LenovoThinkSmart as shown below in the path.
- Copy and paste each of these commands in to the command line:

```
adb root
adb pull /vendor/firmware/wlan/qca_cld/WCNSS_qcom_cfg.ini C:\LenovoThinkSmart
```

- A file named WCNSS_qcom_cfg.ini will be copied from the Lenovo device to C:\LenovoThinkSmart
- Open the file in Windows Notepad
- Locate the line that is:

```
# Each byte of MAC address is represented in Hex format as XX

Intf0MacAddress=000AF58989FF
```

- Change the "FF" at the end of the MAC address to another unique number.
- For example, I use "01" for my first Lenovo device, and "02" for the second, "03 for third, and so on...
- Save the WCNSS_qcom_cfg.ini
- Go back to the ADB and Fastboot ++ program.
- Type "exit" to get back out to the Windows command line and out of ADB shell.
- Copy and paste each of these commands in to the command line:

```
adb root
adb remount
adb push C:\LenovoThinkSmart\WCNSS_qcom_cfg.ini /vendor/firmware/wlan/qca_cld/
```

- You should see an indication that 1 file was successfully uploaded to the Lenovo device......

"C:\LenovoThinkSmart\WCNSS_qcom_cfg.ini: 1 file pushed, 0 skipped. 21.2 MB/s (14421 bytes in 0.001s)"

- Unplug the device.
- Push and hold the VOL + button
- Plug in power adapter to Lenovo device (no Windows sound heard this time)
- Recovery screen should be visible.
- Use VOL + to select "Wipe cache partition"
- Use VOL - to confirm
- Use VOL + to select "Yes"
- Use VOL - to confirm
- Use VOL - to confirm "Reboot system now"
- The Lenovo device will reboot.
- As the device is rebooting, you will see the Donald Duck in the car image.
